crates/devolutions-agent-shared/src/lib.rs

```diff
@@ -11,6 +13,16 @@ use cfg_if::cfg_if;
 pub use date_version::{DateVersion, DateVersionError};
 pub use update_json::{ProductUpdateInfo, UpdateJson, VersionSpecification};
 
+#[cfg(windows)]
+pub fn get_installed_agent_version() -> Result<Option<DateVersion>, windows::registry::RegistryError> {
+    windows::registry::get_installed_product_version(windows::AGENT_UPDATE_CODE)
+}
+
+#[cfg(not(windows))]
+pub fn get_installed_agent_version() -> Result<Option<DateVersion>, Infallible> {
+    Ok(None)
+}
```

Comment:
suggestion: It’s typically safer to not return a different error type to avoid inconsistency (and ending up with cfgs again).